contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/nvidia_drm_modeset_check.py

```diff
@@ -0,0 +1,174 @@
+
+    # Find all card directories (e.g., card0, card1)
+    card_dirs = sorted(
+        [d for d in os.listdir(drm_path) if re.search(r"card[0-9]*$", d)]
+    )
+
+    if not card_dirs:
+        logging.error("No graphics cards found in /sys/class/drm.")
+        return
+
+    logging.info("--- Graphics Card Information ---")
+
+    for card_dir in card_dirs:
+        full_card_path = os.path.join(drm_path, card_dir)
+        device_path = os.path.join(full_card_path, "device")
+
+        connectors = []
+        for item in os.listdir(full_card_path):
+            # Connectors are subdirectories like cardX-DP-1
+            pattern = "{}-".format(card_dir)
+            if item.startswith(pattern):
+                # Extract just the connector name (e.g., DP-1)
+                connector_name = item.replace(pattern, "")
+                connectors.append(connector_name)
+
+        driver = "N/A"
+        uevent_path = os.path.join(device_path, "uevent")
+        if os.path.exists(uevent_path):
+            try:
+                with open(uevent_path, "r") as f:
+                    for line in f:
+                        if line.startswith("DRIVER="):
+                            driver = line.strip().split("=")[1]
+                            break
+            except IOError as e:
+                driver = "Error reading uevent: {}".format(e)
+
+        vendor_name = "Unknown"
+        vendor_path = os.path.join(device_path, "vendor")
+        if os.path.exists(vendor_path):
+            try:
+                with open(vendor_path, "r") as f:
+                    vendor_id = f.read().strip()
+                    vendor_name = VENDOR_MAP.get(
+                        vendor_id, "Unknown ID: {}".format(vendor_id)
+                    )
+            except IOError:
+                vendor_name = "Could not read vendor file"
+
+        card_name = "N/A"
+        if os.path.islink(device_path):
+            try:
+                pci_path = os.path.realpath(device_path)
+                pci_address = os.path.basename(pci_path)
+                if pci_address.split(":")[0].isnumeric():
+                    logging.info(
+                        "Check card name with PCI bus: %s", pci_address
+                    )
+                    lspci_output = subprocess.check_output(
+                        ["lspci", "-s", pci_address], text=True
+                    ).strip()
+                    card_name = " ".join(lspci_output.split(" ")[1:])
+                else:
+                    card_name = "Unknown"
+            except (subprocess.CalledProcessError, FileNotFoundError):
+                card_name = "Unknown"
+
+        card_info = {
+            "Index": card_dir,
+            "Vendor": vendor_name,
+            "Name": card_name,
+            "Driver": driver,
+            "Connectors": connectors,
+        }
+        graphics_info.append(card_info)
+
+    return graphics_info
+
+
+def verify_nvidia_modeset_enable():
+    enable = None
```

Comment:
```suggestion
    enable = None
    modeset = "Unknown"
```
Assign the initial value for `modset` to prevent if the unexpected error from `nvidia_modeset_path` is not exist.
```
raceback (most recent call last):
  File "/home/ceqa/checkbox-provider-ce-oem/bin/./nvidia_drm_modeset_check.py", line 174, in <module>
    main()
  File "/home/ceqa/checkbox-provider-ce-oem/bin/./nvidia_drm_modeset_check.py", line 158, in main
    if not verify_nvidia_modeset_enable():
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ceqa/checkbox-provider-ce-oem/bin/./nvidia_drm_modeset_check.py", line 116, in verify_nvidia_modeset_enable
    if modeset.lower() == "y":
       ^^^^^^^
UnboundLocalError: cannot access local variable 'modeset' where it is not associated with a value
```
Otherwise, LGTM

Reply:
Sideload result on a platform which is not support NV card.
```
ceqa@localhost:~/checkbox-provider-ce-oem/bin$ sudo ./nvidia_drm_modeset_check.py 
2025-09-01 06:04:28 INFO     --- Graphics Card Information ---
2025-09-01 06:04:28 INFO     graphic card information: [{'Index': 'card0', 'Vendor': 'Unknown', 'Name': 'Unknown', 'Driver': 'panfrost', 'Connectors': []}, {'Index': 'card1', 'Vendor': 'Unknown', 'Name': 'Unknown', 'Driver': 'rzg2l-du', 'Connectors': ['HDMI-A-1']}]
2025-09-01 06:04:28 ERROR    Error: no Nvidia graphic card been detected
2025-09-01 06:04:28 ERROR    Error: nvidia-drm.modeset is not enabled
```